### PR TITLE
sweeper for depositing RSC

### PR DESCRIPTION
**Pull request by lightninglu10**
Checks every 5 minutes if we have open deposits and completes them if they are completed

### CODE REVIEW

File: src/reputation/tasks.py [hunk 1]
```diff
@@ -82,6 +82,40 @@ def create_author_contribution(contribution_type, user_id, unified_doc_id, objec
...
+
+
+@periodic_task(
+    run_every=crontab(minute="*/5"),
+    priority=4,
+    queue=QUEUE_PURCHASES,
+)
+def check_deposits():
+    deposits = Deposit.objects.filter(paid_status=None)
+    for deposit in deposits:
```
Comment (le-sun) [inline src/reputation/tasks.py:109]:
```suggestion
    for deposit in deposits.iterator():
```

File: src/reputation/tasks.py [hunk 2]
```diff
@@ -82,6 +82,40 @@ def create_author_contribution(contribution_type, user_id, unified_doc_id, objec
...
+        amt = deposit.amount
+        user = deposit.user
+        transaction_successful = check_transaction_success(deposit.transaction_hash)
+        if transaction_successful:
+            distribution = Dist("DEPOSIT", amt, give_rep=False)
+            distributor = Distributor(distribution, user, user, time.time(), user)
+            distributor.distribute()
```
Comment (le-sun) [inline src/reputation/tasks.py:116]: Should we add a deposit update here to mark it as paid?
```suggestion
            distributor.distribute()
            deposit.set_paid()
```

**Review by le-sun — CHANGES_REQUESTED** [review]

File: src/reputation/tasks.py (hunk 2 above)
Comment (lightninglu10) [reply src/reputation/tasks.py:116]: Oh is the distribute not enough?

Comment (le-sun) [reply src/reputation/tasks.py:116]: The distributor just creates the distribution object and pays it out, but it doesn't update the deposit object.

Comment (lightninglu10) [reply src/reputation/tasks.py:116]: Ah you're right, done

**Review by le-sun — APPROVED** [review]

**Pull request merged**